File: app/src/main/java/com/example/weatherchallengemvvmkotlin/data/network/WeatherClient.java
package com.example.weatherchallengemvvmkotlin.data.network;

import com.example.weatherchallengemvvmkotlin.util.WeatherConstants;
import com.jakewharton.retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;

import javax.inject.Inject;
import javax.inject.Singleton;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

@Singleton
public class WeatherClient extends OkHttpClient {
    private static Retrofit retrofit = null;
    private WeatherNetworkService service;

    @Inject
    public WeatherClient() {
        HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
        interceptor.setLevel(HttpLoggingInterceptor.Level.BASIC);
        OkHttpClient client = new OkHttpClient().newBuilder().addInterceptor(interceptor).build();

        retrofit = new Retrofit.Builder()
                .baseUrl(WeatherConstants.DOMAIN)
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .client(client)
                .build();
    }

    public Retrofit getRetrofit() {
        return retrofit;
    }
}

File: app/src/main/java/com/example/weatherchallengemvvmkotlin/data/network/WeatherClient.kt
package com.example.weatherchallengemvvmkotlin.data.network

import com.example.weatherchallengemvvmkotlin.util.WeatherConstants
import com.jakewharton.retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import io.reactivex.android.schedulers.AndroidSchedulers
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import javax.inject.Inject

class WeatherClient @Inject constructor(): OkHttpClient() {
    private val service: WeatherNetworkService? = null
    val retrofit: Retrofit?
        get() = WeatherClient.Companion.retrofit

    companion object {
        private var retrofit: Retrofit? = null
    }

    init {
        val interceptor = HttpLoggingInterceptor()
        interceptor.level = HttpLoggingInterceptor.Level.BASIC
        val client = OkHttpClient().newBuilder().addInterceptor(interceptor).build()
        WeatherClient.Companion.retrofit = Retrofit.Builder()
                .baseUrl(WeatherConstants.DOMAIN)
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .client(client)
                .build()
    }
}